pkg/storage/etcd/testing/utils.go

```diff
@@ -146,6 +146,7 @@ func (m *EtcdTestServer) waitUntilUp() error {
 func (m *EtcdTestServer) Terminate(t *testing.T) {
 	m.Client = nil
 	m.s.Stop()
+	time.Sleep(250 * time.Millisecond)
```

Comment:
Sure.
Initially, I hoped that this PR won't be necessary and the purpose of it was to check if that helps. But now, since we have so many flakes, I think we need a quick workaround before we fix if properly. Adding a comment.